File: container/hello-r/context/R/greet.R
#' Say hello
#'
#' Pure R function. No argv parsing. Importable from any other R code
#' (a notebook, a test, another package) the same way the Rscript entry
#' calls it.
#'
#' @param name Who to greet.
#' @param shout If TRUE, return the greeting in uppercase.
#' @return The greeting, as a character string.
#' @export
greet <- function(name = "world", shout = FALSE) {
  message <- paste0("Hello, ", name, "!")
  if (isTRUE(shout)) toupper(message) else message
}

File: container/hello-py/context/src/hello_py_cli/greet.py
"""hello-py greet — say hello.

This is the canonical "one verb per file" example. The function below is
both an importable Python function AND the body of the `hello-py greet`
subcommand. Same code, two callers.
"""
import typer


def greet(
    name: str = typer.Option("world", "--name", "-n", help="Who to greet."),
    shout: bool = typer.Option(False, "--shout", help="Use uppercase output."),
) -> None:
    """Print a friendly greeting."""
    message = f"Hello, {name}!"
    typer.echo(message.upper() if shout else message)
